toolbox/bulk_processing/find_invalid_address_case_id_by_uprn.py

```diff
@@ -0,0 +1,77 @@
+import argparse
+import csv
+import os
+from pathlib import Path
+
+import requests
+from google.cloud import storage
+from requests import HTTPError
+
+from toolbox.config import Config
+
+
+def generate_bulk_invalid_address_file(file_to_process):
+    address_delta_file = Path(f'invalid_addresses_{file_to_process.stem}.csv')
+    address_delta_file.write_text('case_id,reason\n')
+    error_count = 0
+
+    with open(file_to_process, encoding="utf-8") as open_file_to_process:
+        file_reader = csv.DictReader(open_file_to_process, delimiter=',')
+
+        with open(address_delta_file, 'a') as file:
```

Comment:
could open the file in regular write mode, because we presume that the file doesn't exist at this point - we certainly wouldn't want to append to an existing file